### PR TITLE
Add Pubkey Metrics To Measure Cache Miss and Hits

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Feature Addition

**What does this PR do? Why is it needed?**

- [x] Adds in pubkey metrics to measure cache miss and hit ratios. In Pyrmont, sync was 50% slower
because our cache wasn't able to hold 100,000 validator pubkeys. Given the low-cost of it, having
a metric to measure hit and miss ratios is useful.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

File: shared/bls/herumi/public_key.go [hunk 1]
```diff
@@ -32,8 +49,11 @@ func PublicKeyFromBytes(pubKey []byte) (common.PublicKey, error) {
 		return nil, fmt.Errorf("public key must be %d bytes", params.BeaconConfig().BLSPubkeyLength)
 	}
 	if cv, ok := pubkeyCache.Get(string(pubKey)); ok {
+		pubkeyCacheHit.Inc()
 		return cv.(*PublicKey).Copy(), nil
 	}
+	pubkeyCacheMiss.Inc()
+
```
Comment (shayzluf) [inline shared/bls/herumi/public_key.go:56]:
```suggestion
```

**Review by shayzluf — COMMENTED** [review]
- it will be nice to know the space left in cache 
- maybe we should already increase the maxKeys to handle pyrmont better

File: shared/bls/blst/public_key.go [hunk 2]
```diff
@@ -7,18 +7,36 @@ import (
 	"fmt"
 
 	"github.com/dgraph-io/ristretto"
+
```
Comment (shayzluf) [inline shared/bls/blst/public_key.go:10]:
```suggestion
```

File: shared/bls/blst/public_key.go [hunk 3]
```diff
@@ -7,18 +7,36 @@ import (
...
-var maxKeys = int64(100000)
-var pubkeyCache, _ = ristretto.NewCache(&ristretto.Config{
-	NumCounters: maxKeys,
-	MaxCost:     1 << 22, // ~4mb is cache max size
-	BufferItems: 64,
-})
+var (
+	pubkeyCacheHit = promauto.NewCounter(
+		prometheus.CounterOpts{
+			Name: "blst_pubkey_cache_hit",
```
Comment (terencechain) [inline shared/bls/blst/public_key.go:22]: Please follow https://prometheus.io/docs/practices/naming/

**Review by terencechain — CHANGES_REQUESTED** [review]
Concern: Wouldn't this increase the cardinality of the metrics page significantly and require tsdb to store much more data than before?

**Comment by prestonvanloon** [comment]
Any update on this?

**Comment by rauljordan** [comment]
Any update on this one @nisdas ?

**Comment by nisdas** [comment]
Not fully relevant now, given that we have a very large pubkey cache. Closing the PR @rauljordan

**Pull request closed without merging**